cdap-client/src/main/java/co/cask/cdap/client/NamespaceClient.java

```diff
@@ -61,9 +62,18 @@ public NamespaceClient(ClientConfig config) {
    * @throws UnAuthorizedAccessTokenException if the request is not authorized successfully in the gateway server
    */
   public List<NamespaceMeta> list() throws IOException, UnAuthorizedAccessTokenException {
-    HttpResponse response = restClient.execute(HttpMethod.GET, config.resolveURL("namespaces"),
-                                               config.getAccessToken());
-    return ObjectResponse.fromJsonBody(response, new TypeToken<List<NamespaceMeta>>() { }).getResponseObject();
+    // TODO: remove the following apiVersion set/reset logic all APIs are migrated to v3
```

Comment:
Yes, this sets api version in ClientConfig. However, the problem is that we can't set the API version once and leave it, because not all resources have v3 APIs yet. So I thought its better to let the client dictate what version to use. To support multiple of them it may be tricky. Thoughts?